filecoin-proofs/src/api/internal.rs

```diff
@@ -199,17 +203,50 @@ const SLOTH_ITER: usize = 0;
 const LAYERS: usize = 4; // TODO: 10;
 const TAPER_LAYERS: usize = 2; // TODO: 7
 const TAPER: f64 = 1.0 / 3.0;
-const CHALLENGE_COUNT: usize = 2;
+const POREP_MINIMUM_CHALLENGES: usize = 12; // FIXME: 8,000
+
 const DRG_SEED: [u32; 7] = [1, 2, 3, 4, 5, 6, 7]; // Arbitrary, need a theory for how to vary this over time.
 
-lazy_static! {
-    static ref CHALLENGES: LayerChallenges =
-        LayerChallenges::new_tapered(LAYERS, CHALLENGE_COUNT, TAPER_LAYERS, TAPER);
+fn select_challenges(
+    partitions: usize,
+    minimum_total_challenges: usize,
+    layers: usize,
+    taper_layers: usize,
+    taper: f64,
+) -> LayerChallenges {
+    let mut count = 1;
+    let mut guess = LayerChallenges::Tapered {
+        count,
+        layers,
+        taper,
+        taper_layers,
+    };
+    while partitions * guess.total_challenges() < minimum_total_challenges {
+        count += 1;
+        guess = LayerChallenges::Tapered {
+            count,
+            layers,
+            taper,
+            taper_layers,
+        };
+    }
+    guess
 }
 
-fn setup_params(sector_bytes: PaddedBytesAmount) -> layered_drgporep::SetupParams {
+fn setup_params(
+    sector_bytes: PaddedBytesAmount,
+    partitions: usize,
```

Comment:
SUGGESTION: I see that this function calls `select_challenges` with `POREP_MINIMUM_CHALLENGES`. I see that it has a `partitions` parameter which is of type `usize`. I think it makes sense to use `PoRepProofPartitions` as the type for the `partitions` parameter instead.

Reply:
`select_challenges` is purely about calculating a `LayerChallenges` that meets some challenge requirement. It could be logically housed in `filecoin_proofs` or used to reason about hypothetically (but not yet) supported partition amounts. (For example, see the unit test used to gain confidence in the correctness of its output.) If `select_challenges` were to take a`PoRepProofPartitions` we would still want another function exactly like the current one that it could call to do the real work. That seems like yet another layer of indirection I'd rather avoid.